test/prober.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// route.go provides methods to perform actions on the route resource.
+
+package test
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/pkg/test/spoof"
+	"golang.org/x/sync/errgroup"
+)
+
+// Prober is the interface for a prober, which checks the result of the probes when stopped.
+type Prober interface {
```

Comment:
I see. If we're going to have more probers, then sure. Though, it seems go's suggestion to define interface in the client than in the library seems better (since perhaps for conformance tests there might more/less methods).

Reply:
With my most recent changes, ProberManager now also satisfies Prober :)